includes/multi-currency/Compatibility.php

```diff
@@ -87,12 +87,32 @@ public function override_selected_currency() {
 	}
 
 	/**
-	 * Checks to see if the widgets should be hidden.
+	 * Deprecated method, please use should_disable_currency_switching.
 	 *
 	 * @return bool False if it shouldn't be hidden, true if it should.
 	 */
 	public function should_hide_widgets(): bool {
-		return apply_filters( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', false );
+		return $this->should_disable_currency_switching( apply_filters( MultiCurrency::FILTER_PREFIX . 'should_hide_widgets', false ) );
```

Comment:
I feel like we should add some sort of deprecated notice logging here.

Reply:
Added in https://github.com/Automattic/woocommerce-payments/pull/7152/commits/7d9a3f2dfa03e89c52819d4582b9b270e90f90ba